### PR TITLE
server: Allow filtering on org_id from the identity header

### PR DESCRIPTION
**Pull request by Gundersanne**
For now when the environment is empty the filter won't apply.

### CODE REVIEW

File: internal/server/server.go [hunk 1]
```diff
@@ -270,18 +273,47 @@ func (b binder) Bind(i interface{}, ctx echo.Context) error {
...
+	} `json:"identity"`
+}
+
+// clouddot guidelines requires 404 instead of 403 when unauthorized
+func (s *Server) VerifyIdentityHeader(nextHandler echo.HandlerFunc) echo.HandlerFunc {
 	return func(ctx echo.Context) error {
 		request := ctx.Request()
 
 		// For now just check it's there, in future we might want to
 		// decode the b64 string and check a specific entitlement
```
Comment (teg) [inline internal/server/server.go:290]: We decode it now, so drop the comment?

**Review by teg — CHANGES_REQUESTED** [review]
Love it! This is exactly what we need :) One nitpick, and there were a couple of warnings about missing tests. Could you either add the tests, or add a comment to the commit message to say why not?

**Comment by Gundersanne** [comment]
@teg I added a bunch of unit tests. The integration tests only test the expect path. Do you want me to replicate the unit tests in the integration tests?

**Comment by teg** [comment]
Nah, no need to replicate the tests. I'm totally fine with saying "the warnings are moot because of reasons". Just thought I'd ask since we have them enabled ;)

**Comment by Gundersanne** [comment]
Oh, i tuned those out automatically, probably a bad reflix! Hm I could easily add a test for one of those. I'll do that and then ask for a rereview :)

**Comment by teg** [comment]
> Oh, i tuned those out automatically, probably a bad reflix! Hm I could easily add a test for one of those. I'll do that and then ask for a rereview :)

Hehe, I know the reflex. Reviewers are here to tune them back in :P

**Review by teg — APPROVED** [review]
Great! Thanks for this!

**Pull request merged**